doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -1724,9 +1724,53 @@ Ltac2 features a toplevel loop that can be used to evaluate expressions.
    together with its type. This command is pure in the sense that it does not
    modify the state of the proof, and in particular all side-effects are discarded.
 
+Compilation
+-----------
+
+Ltac2 definitions can be compiled to OCaml, the compiled definitions
+will then run natively instead of using the interpreter.
+
+.. cmd:: Ltac2 Compile {* @qualid }
+
+   Compile the given Ltac2 definitions. Compilation does not survive
+   state resets (end of file, end of module, end of section).
+
+   Ltac2 code in an :ref:`antiquotation <term-antiquotations>` will
+   not be compiled. It will also not be seen by dependency analysis,
+   so Ltac2 definitions appearing in an antiquotation which appears in
+   an argument of `Ltac2 Compile` will not be automatically compiled.
+
+   Compiled code currently does not respect :flag:`Ltac Profiling` and
+   :flag:`Ltac2 Backtrace`.
```

Comment:
I expect it will not play nicely with the debugger, either, when that becomes available.  But not an issue for today.